### ISSUE TITLE
[Goldsrc] Higher up the gl_max_size default value

### ISSUE THREAD

**Issue by GBCS-PRO** [ValveSoftware/halflife#2459]
I don't know why it's still capped to 256p even though the highest supported resolution reaches 512p and sometimes reach resolutions like 786x384.
HQ custom maps/models would look so bad if the value wasn't manually set to 512 or 1024. Now with PO2 scale down issue being fixed, this should also be changed.

**Comment by 10th** [comment]
512 - default value
256 - minimum value

**Comment by GBCS-PRO** [comment]
No the default is set to 256 in all Goldsrc titles..

**Comment by drogga** [comment]
> No the default is set to 256 in all Goldsrc titles..

The value of gl_max_size is by default 512 and depends on the capabilities of your video card:

![max_size](https://user-images.githubusercontent.com/43634578/58277831-7d532480-7d4f-11e9-8da6-354744a80764.PNG)

The correct value for your video card is taken from:
C:\Program Files\Steam\steamapps\common\Half-Life\valve\hw\[your video card].cfg

![max_size2](https://user-images.githubusercontent.com/43634578/58277838-8512c900-7d4f-11e9-9053-4cd3d4a7a4fb.PNG)

For the nVidia GeForce video card (geforce.cfg) gl_max_size "512" by default.

**Comment by GBCS-PRO** [comment]
I used to have Geforce gpu and it was defaulted to 256 only. I dunno what's the case here

**Comment by SamVanheer** [comment]
It depends on what the OpenGL driver reports as the renderer, which isn't necessarily your GPU if it's using the integrated card or if some hack/workaround/etc is causing it to report something else.

**Comment by mikela-valve** [comment]
@10th's suggestions on #2374 are also worth considering, if anyone else has any comments on that.

**Comment by GBCS-PRO** [comment]
@mikela-valve I totally agree with points there. But let's have the models cap higher up to 1024 (to make textures over 512p like 786x384 more useful and not scaled down..)

**Comment by djdallmann** [comment]
Default is 512 for me unless I have low vid quality mode set in cs 1.6 video settings for example.

**Comment by ChaosSpoofer** [comment]
I just noticed that in the beta, GoldSrc games now seem hardwired to gl_max_size 512, with no way to change it.  You can alter the value to 256 in standard configs, hw/(graphics card/driver).cfg, or in console before first rendering the game, and you'll still always see relevant textures rendered at 512.

I switched back to the non-beta, and gl_max_size actually works again, where I can switch gl_max_size back to 256 if I wish and I'll see the differences in the textures.

**Comment by SirYodaJedi** [comment]
> I just noticed that in the beta, GoldSrc games now seem hardwired to gl_max_size 512, with no way to change it. You can alter the value to 256 in standard configs, hw/(graphics card/driver).cfg, or in console before first rendering the game, and you'll still always see relevant textures rendered at 512.
> 
> I switched back to the non-beta, and gl_max_size actually works again, where I can switch gl_max_size back to 256 if I wish and I'll see the differences in the textures.

This appears to have since been pushed to the standard build. This is a bit of a problem, because, as OP mentions, textures wider or taller than 512tx are technically valid, as long as you don't exceed 512² total texels. Apparently 1024x512 was valid back in 2018 (according to some old TWHL discord messages)?

And yeah, there really aren't any GPUs that can run Steam Half-Life that struggle with 512x512 textures, and they all can technically handle higher-res textures; if someone wants to use a Voodoo 2 to play HL1, they'll be running a retail copy. It's probably safest to set the default `gl_max_size` to 1024, since higher resolutions are an anomaly.
